src/task.c

```diff
@@ -711,6 +719,12 @@ JL_DLLEXPORT void jl_switch(void) JL_NOTSAFEPOINT_LEAVE JL_NOTSAFEPOINT_ENTER
     if (other_defer_signal && !defer_signal)
         jl_sigint_safepoint(ptls);
 
+    if (ptls->wait_in_scheduler) {
+        ptls->wait_in_scheduler = 0;
+        jl_set_next_task(jl_task_get_next());
```

Comment:
@vtjnash: how would we get the various arguments needed for `jl_task_get_next` here?

Reply:
That seems quite tedious. I think https://github.com/JuliaLang/julia/pull/57544 is probably the better direction to go